services/ml/execution_impl_ie.cc

```diff
@@ -31,21 +42,52 @@ int32_t ExecutionImplIe::Init(int32_t preference) {
       device_name = "GPU";
     } else if (preference == mojom::PREFER_LOW_POWER) {
       device_name = "MYRIAD";
+    } else if (preference == mojom::PREFER_ULTRA_LOW_POWER) {
+      device_name = "GNA";
+      // Release in squence to avoid crash. Close GNA device befere re-open,
+      s_gna_infer_request.reset(nullptr);
+      s_gna_execution.reset(nullptr);
+      s_gna_plugin.reset(nullptr);
     }
     DLOG(INFO) << "[IE] Trying to get plugin by device name " << device_name;
-    plugin_.reset(
+    std::unique_ptr<InferenceEngine::InferRequest> infer_request;
+    std::unique_ptr<InferenceEngine::InferencePlugin> plugin;
+    std::unique_ptr<InferenceEngine::ExecutableNetwork> execution;
+    plugin.reset(
         new ie::InferencePlugin(static_cast<ie::InferenceEnginePluginPtr>(
-            ie::PluginDispatcher({""}).getPluginByDevice(device_name))));
-    const ie::Version* version = plugin_->GetVersion();
+            ie::PluginDispatcher({
+#if defined(OS_WIN)
+              L""  // Windows support UNICODE.
+#else
+              ""
+#endif
+            }).getPluginByDevice(device_name))));
+    const ie::Version* version = plugin->GetVersion();
     DLOG(INFO) << "[IE] succeed to load plugin " << version->buildNumber << " "
                << version->description;
-    execution_.reset(
+    std::map<std::string, std::string> plugin_Config = {};
+    if (preference == mojom::PREFER_ULTRA_LOW_POWER && input_scale_ > 0) {
+      std::string scaleFactorConfigKey = GNA_CONFIG_KEY(SCALE_FACTOR);
+      plugin_Config[scaleFactorConfigKey] = std::to_string(input_scale_);
+      // gnaPluginConfig[ie::GNAConfigParams::KEY_GNA_PRECISION] = "I16";
```

Comment:
do we need this line?

Reply:
We can also use int8 to inference with the configure, i will confirm whether use int8 by default.

Reply:
How about keeping the line until GNA version 3 [seeing the comments](https://github.com/opencv/dldt/tree/2020/inference-engine/samples/speech_sample#gna-specific-details)?

> Note that it is not always possible to use 8-bit weights due to GNA hardware limitations. For example, convolutional layers always use 16-bit weights (GNA harware verison 1 and 2). This limitation will be removed in GNA hardware version 3 and higher.

Reply:
I am fine to keep it with proper comments. Please add them. Thanks!